### DISCUSSION TITLE
[Bug ?] Conflict between 2 cosmetic filters

### DISCUSSION THREAD

**Discussion by Procyon-b** [uBlockOrigin/uBlock-issues#3480] [General] [answered]
https://new.reddit.com/r/uBlockOrigin/comments/1hbit6l/elementsads_on_raiderio_refuse_to_stay_blocked/

Yesterday a user came on reddit with a problem with a userfilter that isn't applied by uBO. I was unable to reproduce, and since the user has enabled many extra lists (43) I was unable to continue debugging. He later came back after identifying the list which was causing the issue ("_AdGuard – Social Widgets_").

website: raider.io
userfilter: `raider.io##.slds-size--1-of-1.slds-col > div.fresnel-greaterThan-sm.fresnel-container`
  targets at least 2 blocks on the index page (The top one contains "New: Recruitment system")

Test:
Disabling all lists except "_My filters_" and "_AdGuard – Social Widgets_", enable "_Ignore generic cosmetic filters_", add the userfilter and reload the page. The top block isn't hidden.

Using `raider.io##.slds-size--1-of-1.slds-col > div.fresnel-greaterThan-sm.fresnel-container:style(opacity: .2; outline: 2px solid red;)` instead marks the block correctly.

Using  `raider.io##.slds-size--1-of-1.slds-col > div.fresnel-greaterThan-sm.fresnel-container:style(display: none;)` also works as expected.

Only 3 cosmetic filters in the list are targeting this site. And one is responsible. After simplification, it is:
`raider.io##.twitter-timeline|blog.jp`

Now, disable the list, and run uBO with only that filter and the user's filter. The userfilter won't be applied, but styling the target works.

TLDR: 
a (hiding) cosmetic filter prevents hiding an element with another (hiding) userfilter
but you can style the target element with `:style()`
and you can even hide it with `:style(display: none;)`

**Comment by gorhill** [comment]
I will need to investigate more but my first guess is that there is an invalid cosmetic filter in the AdGuard list, undetected by uBO, which breaks cosmetic filtering.

**Reply by Yuki2718** [reply]
It looks to be a typo in https://github.com/AdguardTeam/AdguardFilters/commit/051ef5aabaab333fa6ab21d5df869f9a1d4aa6d9
Corrected https://github.com/AdguardTeam/AdguardFilters/commit/6e2aa19030c692404944582fd8b9fea075b6c57e

**Reply by gorhill** [reply]
The selector is seen as valid by uBO since `|` is a valid [namespace operator](https://developer.mozilla.org/en-US/docs/Web/CSS/Namespace_separator).

**Reply by Procyon-b** [reply]
@gorhill but why does it make the other filter irrelevant ?  Is it only a browser bug ?

**Reply by gorhill** [reply, marked as answer]
Because all plain CSS cosmetic filters are injected as a single CSS declaration. If one selector is bad, the whole CSS declaration is bad and ignored by the browser.

**Reply by Procyon-b** [reply]
Edit:
scratch that. The 2 cents just dropped.  ;)
/edit:

~~OK. I assume that the filters with :style() are injected in a separate declaration ?~~
Thanks

**Comment by uBlock-user** [comment]
>  raider.io##.twitter-timeline|blog.jp

invalid filter

**Reply by Procyon-b** [reply]
I know that it's an invalid filter. But why does it prevent uBO to use the other filter (except when :style() is used) ?

**Reply by uBlock-user** [reply]
![Capture](https://github.com/user-attachments/assets/9282fd44-e087-42d6-b03a-270c987d1324)

@gorhill uBO is unable to handle syntax error gracefully it seems

**Reply by uBlock-user** [reply]
@Procyon-b If you remove the `|blog.jp` part, from the filter then the issue doesn't occur

**Reply by gorhill** [reply]
> unable to handle syntax error gracefully it seems

It's from the logger related-code specifically. The error can be caught, and might be good to report the error to the logger so that it becomes easier to spot invalid selectors.

**Reply by u-RraaLL** [reply]
Just referencing the previous case/discussion so it's easier to find them in the future: https://github.com/uBlockOrigin/uBlock-discussions/discussions/833#discussioncomment-8669882
